### ISSUE TITLE
Service weights should also apply to connect mesh

### ISSUE THREAD

**Issue by radykal-com** [hashicorp/consul#9334]
#### Feature Description

When registering services (as per https://www.consul.io/docs/discovery/services#dns-srv-weights ) you are allowed to set service weights so clients can use them to balance the requests accurately to the destination capacity. Weights work fine when using consul DNS interface but not when using connect mesh. As consul register the sidecar proxies with default weight values of 1 and 1 (for passing and warning).

#### Use Case(s)

As described in this forum post https://discuss.hashicorp.com/t/weighted-round-robin-with-consul-connect/18526/2 when your upstream servers are not of the same size (using AWS spot instances for example) and you have different number of CPU/RAM/Network resources, it would be great to allow sidecar proxies to also use the weights.

**Comment by banks** [comment]
@radykal-com thanks for this contribution!

As I think you figured out, we do already populate the weights correctly in Envoy, however they are done based on the _proxy's_ registration not the original service's one.

So it's already possible to use that feature today if you explicitly set them on the [sidecar definition](https://www.consul.io/docs/connect/registration/sidecar-service) not the main service one:

```hcl
service {
  name = "foo"
  connect {
    sidecar_service {
      weights = {
        passing = 10
      }
    }
  }
}
```

That said, your PR which makes the sidecar weights default to the same as the service weights makes a lot of sense too. Thanks for that!

**Comment by radykal-com** [comment]
@banks thanks you, I didn't know it was posible to define them in the sidecar definition. So to have a consistent weights across protocols (mesh, dns, etc) the best way is to define them on both services right?

**Comment by banks** [comment]
> So to have a consistent weights across protocols (mesh, dns, etc) the best way is to define them on both services right?

Yes that makes sense to me although typically once you are using a mesh with sidecar proxies then resolving the non-mesh IPs with DNS (i.e. `foo.service.consul` rather than `foo.connect.consul` is not very useful since your applications are likely not accessible unless you go through the mesh.

I think your PR makes sense though just from a UX standpoint - it saves the nesting shown above and is consistent with how we default things for other fields. It also helps during migration so the weights you might have configured for DNS before you adopted mesh carry over without additional effort.
